mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -239,10 +239,7 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
     }
     val familyAndLink = new FamilyAndLink(familyObj, linkObj)
 
-    val numFeatures = dataset.select(col($(featuresCol))).limit(1).rdd
-      .map { case Row(features: Vector) =>
-        features.size
-      }.first()
+    val numFeatures = dataset.select(col($(featuresCol))).first().getAs[Vector](0).size
```

Comment:
Ah right, we would need to add an implicit encoder `implicit def encoder: Encoder[Vector] = ExpressionEncoder()`, e.g. see here https://github.com/apache/spark/pull/12718#issuecomment-215332276.

However, let's leave that change for #12718